File: 1046-MaxConsecutiveOnesIii/1046-MaxConsecutiveOnesIii.py
# Last updated: 07/04/2026, 11:37:28
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        out = 0 
        ls =[]
        i = 0 
        l = 0 
        pos = -1
        for j in range(len(nums)):
            if nums[j] == 0 :
                ls.append(j)
                l+=1
            if nums[j] == 0 and l >k :
                pos = ls[i]
                i+=1
                l-=1
            

            out  = max(out,j-pos)
        return out

File: 1046-MaxConsecutiveOnesIii/1046-MaxConsecutiveOnesIii.js
// Last updated: 07/04/2026, 11:37:23
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var longestOnes = function(nums, k) {
    let i= 0 ;
    let zero = [];
    let ans = 0 
    let pos = -1 
    let countK = 0

    for (let j = 0 ;j<nums.length;j++){
        if (nums[j] === 0 ){
            countK++;
            zero.push(j);
        }
        if (nums[j] === 0  && countK> k ){
            pos = zero[i];
            i++;

        }
        ans = Math.max(ans,j-pos);
    }
    return ans 
};